### PR TITLE
DOC-5902 - edit revs_limit description

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-5902

- Edit first sentences
- Add title for advanced topic about disconnected branches

### CODE REVIEW

**Review by ibsoln — APPROVED** [review]

File: modules/ROOT/assets/attachments/sg.yaml [hunk 1]
```diff
@@ -295,22 +295,25 @@ properties:
...
-            
-              If there are conflicting revisions, the document may end up with **disconnected branches** after the pruning process. In the animation below, the document has a conflicting branch (revisions `4'` - `1001'`). When the shortest branch (in this case the conflicting branch) reaches the 1003rd update, it gets is cut off. The revision tree is not in a corrupted state and the logic that chooses the winning revision still applies. But it may make it impossible to do certain merges (n-way merge) to resolve conflicts and will occupy disk space that could have been free-ed if the conflict was resolved early on.<br><br>
-            
+              Maximum depth to which a document's revision tree can grow, it governs the point at which to prune a document's revision tree. Revision pruning is the process to remove obsolete parent revisions and runs every time a revision is added.
+
+              The minimum value is 20.
```
Comment (rajagp) [inline modules/ROOT/assets/attachments/sg.yaml]:
```suggestion
              The minimum and default values that this property can take is dependent on the version of Sync Gateway 
```

Comment (rajagp) [reply modules/ROOT/assets/attachments/sg.yaml]: Please decide on best way to present this 
**Sync Gateway version 2.6+** 
The value of the  [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) config setting
- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is `true` then the default value is 100 and min value is 20
- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is `false` then the default value is 50 and min value is 1

**Sync Gateway version 2.0 to 2.6** 
The value of the  [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) config setting
- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is `true` then the default value is 1000 and min value is 20
- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is `false` then the default value is 1000 and min value is 1

**Sync Gateway version 1.x** 
-  the default value is 1000 and min value is 20

File: modules/ROOT/assets/attachments/sg.yaml [hunk 2]
```diff
@@ -295,22 +295,25 @@ properties:
...
+
+              ##### Consideration with Conflicting Revisions
+
```
Comment (rajagp) [inline modules/ROOT/assets/attachments/sg.yaml:303]: The discussion below is relevant when [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) is true in which case, conflicts are allowed in the system.

File: modules/ROOT/assets/attachments/sg.yaml [hunk 3]
```diff
@@ -295,22 +295,25 @@ properties:
...
+              Setting `revs_limit` to a low value (typically below 100) can have adverse effects with conflicting revisions.
```
Comment (rajagp) [inline modules/ROOT/assets/attachments/sg.yaml]:
```suggestion
              The value of `revs_limit` has implications on how conflict resolution.  Setting it to a very low value can have adverse effects with conflicting revisions as discussed below. This is the reason for a min value requirement on this configuration
```

File: modules/ROOT/assets/attachments/sg.yaml (hunk 2 above)
Comment (ibsoln) [reply modules/ROOT/assets/attachments/sg.yaml:303]: Thank-you

Comment (ibsoln) [reply modules/ROOT/assets/attachments/sg.yaml:303]: Thanks-you

**Review by ibsoln — APPROVED** [review]

File: modules/ROOT/assets/attachments/sg.yaml [hunk 4]
```diff
@@ -295,22 +295,25 @@ properties:
...
+
+              If the revision tree gets into this state then the only option to resolve the conflict is to pick a winning branch and tombstone all the non-winning conflicting branches. Tuning the `revs_limit` to a lower value increases the likelyhood of this scenario happening.
+
               See also:
-              - Couchbase Lite [pruning](../../couchbase-lite/native-api/revision/index.html#pruning).
-              - Sync Gateway purge endpoint [/{db}/_purge](references/sync-gateway/admin-rest-api/index.html#!/document/post_db_purge).
-              - Sync Gateway [document TTLs](references/sync-gateway/admin-rest-api/index.html#!/document/put_db_doc).
+              - Sync Gateway purge endpoint [/{db}/_purge](admin-rest-api.html#/document/post__db___purge).
+              - Sync Gateway [document TTLs](admin-rest-api.html#/document/put__db___doc_).
             default: 1000
```
Comment (rajagp) [inline modules/ROOT/assets/attachments/sg.yaml]: Please update default: 1000  based on earlier comment on default and min values.

**Comment by rajagp** [comment]
+ @ibsoln  I seemed to have lost my commit comments . I am adding them here to avoid losing them again 
**Sync Gateway 2.6+**

- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is true then the default value is 100 and min value is 20
- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is false then the default value is 50 and min value is 1

**Sync Gateway 2.0 - 2.5**

- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is true then the default value is 1000 and min value is 20
- If [allow-conflicts](https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts) property is false then the default value is 1000 and min value is 1

**Sync Gateway 1.x*

- The default value is 1000 and min value is 20.  The allow-conflicts does not apply.

File: modules/ROOT/assets/attachments/sg.yaml (hunk 4 above)
Comment (rajagp) [reply modules/ROOT/assets/attachments/sg.yaml]: Also include reference to https://docs.couchbase.com/sync-gateway/2.6/config-properties.html#databases-foo_db-allow_conflicts

**Pull request merged**